sds_data_manager/lambda_code/batch_starter.py

```diff
@@ -338,6 +336,60 @@ def extract_components(filename: str):
     return components
 
 
+def send_lambda_put_event(command_parameters):
+    """Sends custom PutEvent to EventBridge
+
+    Example of what PutEvent looks like:
+    event = {
+        "Source": "imap.lambda",
+        "DetailType": "Job Started",
+        "Detail": {
+            "file_to_create": "<s3_uri>",
+            "status": "INPROGRESS",
+            "dependency": "[{
+                "codice": "s3-test",
+                "mag": "s3-filepath"
+            }]")
+        },
+    }
+
+    Parameters
+    ----------
+    command_parameters : str
+        IMAP cli command input parameters.
+        Example of input:
+            "--instrument codice
+            --level l1a
+            --s3_uri '<s3-filepath>'
+            --dependency 'list of dict'"
+    Returns
+    -------
+    dict
+        EventBridge response
+    """
+    event_client = boto3.client("events")
+
+    # Get event inputs ready
+    command = command_parameters.split("--")
+    s3_uri = command[3].replace("s3_uri '", "").replace("' ", "")
+    dependency = command[-1].replace("dependency ", "")
+
+    # Create event["detail"] information
+    detail = EventDetail(
+        file_to_create=s3_uri,
+        status=models.Status.INPROGRESS.value,
+        dependency=dependency,
+    )
```

Comment:
I will just voice my slight preference for not using dataclasses everywhere and leaving this as a simple dictionary and avoiding the `asdict(EventDetail)` call in the other dataclass. I know I'm in the minority here though, so it is fine to keep it as-is too.
```suggestion
    detail = {
        "file_to_create": s3_uri,
        "status": models.Status.INPROGRESS.value,
        "dependency": dependency,
    }
```